bookkeeper-server/src/test/java/org/apache/bookkeeper/client/api/BookKeeperApiTest.java

```diff
@@ -275,11 +276,13 @@ public void testCannotDeleteLedgerTwice() throws Exception {
         result(newDeleteLedgerOp().withLedgerId(lId).execute());
     }
 
-    private static void checkEntries(Iterable<LedgerEntry> entries, byte[] data)
+    private static void checkEntries(LedgerEntries entries, byte[] data)
         throws InterruptedException, BKException {
-        for (LedgerEntry le : entries) {
-            assertArrayEquals(data, le.getEntry());
+        Iterator<LedgerEntry> iterator = entries.retainIterator();
```

Comment:
I don't think you need retainIterator here.

Reply:
Thanks. right, it is now underneath LedgerEntryImpl, so below line entry.getEntry() will not release the reference now. will change it.